kpops/components/common/kubernetes_model.py

```diff
@@ -1,12 +1,87 @@
 import enum
+from collections.abc import Iterator
+from typing import Any
 
-from pydantic import Field
+import yaml
+from pydantic import ConfigDict, Field
+from typing_extensions import override
 
 from kpops.utils.docstring import describe_attr
-from kpops.utils.pydantic import DescConfigModel
+from kpops.utils.pydantic import CamelCaseConfigModel, DescConfigModel
 
 # Matches plain integer or numbers with valid suffixes: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory
 MEMORY_PATTERN = r"^\d+([EPTGMk]|Ei|Pi|Ti|Gi|Mi|Ki)?$"
+K8S_LABEL_MAX_LEN = 63
+
+
+class ManagedFieldsEntry(CamelCaseConfigModel):
+    fields: dict[str, Any] | None = None
+
+
+class OwnerReference(CamelCaseConfigModel):
+    apiVersion: str
+    kind: str
+    name: str
+    uid: str
+    controller: bool | None = None
+    block_owner_deletion: bool | None = None
+
+
+class ObjectMeta(CamelCaseConfigModel):
+    """Metadata for all Kubernetes objects.
+
+    ref: https://gtsystem.github.io/lightkube-models/1.19/models/meta_v1/#objectmeta
+
+    """
+
+    annotations: dict[str, str] | None = None
+    creation_timestamp: str | None = Field(
+        default=None, description="Timestamp in RFC3339 format"
+    )
+    deletion_grace_period_seconds: int | None = None
+    deletion_timestamp: str | None = Field(
+        default=None, description="Timestamp in RFC3339 format"
+    )
+    finalizers: list[str] | None = None
+    generate_name: str | None = None
+    generation: int | None = None
+    labels: dict[str, str] | None = None
+    managed_fields: list[ManagedFieldsEntry] | None = None
+    name: str | None = None
+    namespace: str | None = None
+    owner_references: list[OwnerReference] | None = None
+    resource_version: str | None = None
+    self_link: str | None = Field(
+        default=None,
+        description="Deprecated field, not populated by Kubernetes in modern versions",
+    )
+    uid: str | None = None
+
+    model_config = ConfigDict(extra="allow")
+
+
+class KubernetesManifest(CamelCaseConfigModel):
+    api_version: str
+    kind: str
+    metadata: ObjectMeta
+
+    model_config = ConfigDict(extra="allow")
+
+    @classmethod
+    def from_yaml(
+        cls, /, content: str
+    ) -> Iterator["KubernetesManifest"]:  # TODO: typing.Self for Python 3.11+
+        manifests: Iterator[dict[str, Any]] = yaml.load_all(content, yaml.Loader)
+        for manifest in manifests:
+            yield cls(**manifest)
+
+    @override
+    def model_dump(self, **_: Any) -> dict[str, Any]:
+        return super().model_dump(
+            mode="json",
+            by_alias=True,
+            exclude_unset=True,
```

Comment:
@disrupted 
I was thinking about the `exclude_unset` here. This can cause problems when we define a model lime `StrimziKafkaTopic` from the `KubernetesManifest` class.

Imagine the model is defined as follows:

```python
class StrimziKafkaTopic(KubernetesManifest):
	api_version: str = "v1/beta2"
	kind = "KafkaTopic"
	
	# Rest of the fields...
	spec = TopicSpec
	metadata: ObjectMeta
```

If I create a model and dump it. The `exclude_unset` would not dump the two fields `api_version` and `kind`. 

If we remove that one, all the noise from the ObjectMeta will be dumped with it.

Reply:
good point, I think we would need a custom `@model_serializer` then. Afaict there is no builtin way to specify which default values to always include when `exclude_unset=True`